core/src/main/java/com/github/mizool/core/concurrent/BufferedStreamAdapter.java

```diff
@@ -47,21 +52,82 @@
  * Any {@link Throwable} thrown by one of the incoming futures is wrapped in an {@link UncheckedExecutionException} and
  * thrown while supplying the new stream with values.<br>
  * <br>
- * This buffer is intended to be used like so:<br>
+ * This class is intended to be used as follows:<br>
+ * <pre>{@code
+ * Stream<Completable<V>> completables;
+ * Stream<V> values2 = BufferedStreamAdapter.completable().adapt(completables, bufferSize, executorService);}</pre>
  * <pre>{@code
- * Stream<ListenableFuture<V>> futureStream;
- * Stream<V> resultStream = BufferedStreamAdapter.adapt(futureStream, bufferSize, executorService);}</pre>
+ * Stream<ListenableFuture<V>> listenables;
+ * Stream<V> values1 = BufferedStreamAdapter.listenable().adapt(listenables, bufferSize, executorService);}</pre>
  */
-public final class BufferedStreamAdapter<V>
+public final class BufferedStreamAdapter<F, V>
 {
+    @NoArgsConstructor(access = AccessLevel.PRIVATE)
+    public static class Listenable
+    {
+        public <V> Stream<V> adapt(
+            Stream<ListenableFuture<V>> futures, int bufferSize, ExecutorService executorService)
+        {
+            verifyBufferSize(bufferSize);
+
+            return new BufferedStreamAdapter<ListenableFuture<V>, V>(futures,
+                bufferSize,
+                executorService,
+                this::addListener).adapt();
+        }
+
+        private <E> void addListener(ListenableFuture<E> future, Listener<E> listener)
```

Comment:
Actually, I wanted to avoid reusing either F and V (the type parameters of BufferedStreamAdapter) anywhere - if a nested class or its members use the same letter, it will shadow the outer one, which is confusing and/or misleading.

I'll make this consistent by changing the V to an E in line 68.

Reply:
Hrm... I was on the brink of asking you to make the inner classes non-static, so you can use the actual `V` from the outer class. However, I don't like non-static inner classes either. So yeah, consistent non-shadowing naming it is.